File: C/Assignment26/Program3.c
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//  
// File name :   Program3.c
// Description : Write a program which accept string & a character from user. Return index of first occurence of that character.
// Author :      Pallavi Omprakash Malewar
// Date :        22/07/2026
//
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


/*

Input :     "Marvellous Multi OS"
            M

Output :    0

Input :     "Marvellous Multi OS"
            e

Output :    4

*/

#include<stdio.h>
#include<stdlib.h>

int FirstChar(char *str, char ch)
{
    int index = 0;

    while(*str != '\0')
    {
        if(*str == ch) return index;
        str++;
        index++;
    }
    return -1;
}

int main()
{
    char arr[20];
    char cValue;
    int iRet = 0;

    printf("Enter the String : ");
    scanf("%[^'\n']s", arr);

    printf("Enter the character : ");
    scanf(" %c", &cValue);

    iRet = FirstChar(arr, cValue);

    if(iRet == -1)
    {
        printf("Character is not present.\n");
    }
    else
    {
        printf("Character location is %d\n", iRet);
    }

    return 0;
}

File: JAVA/Assignment 9/Program3.java
/////////////////////////////////////////////////////////////////////////////////////////
//                                                                                     //
// Problem Statement : Write Java program which accept array of characters from user & //
//                     accept one character. Return occurence of that character without//
//                     considering case.                                               //
// Author : Pallavi Omprakash Malewar                                                  //
// Date : 3 Aug 2026                                                                   //
//                                                                                     //
/////////////////////////////////////////////////////////////////////////////////////////


/*  

    Input : b   N   e   B   R   b   A   i   G   i
            b

    Output: 3

*/


import java.util.Scanner;

class MyArray
{
    public int Search(char Arr[], char ch)
    {
        int n = Arr.length;
        int i = 0, iCount = 0;

        for(i = 0 ; i < n ; i++)
        {
            if(Arr[i] <= 'Z' && Arr[i] >= 'A') iCount++;
            if(Arr[i] == ch) iCount++;
        }
        return iCount;
    }
}

class Program3
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.print("Enter the Size of Array : ");
        int No = sobj.nextInt();

        char Arr[] = new char[No];

        for(int i = 0 ; i < No ; i++)
        {
            Arr[i] = sobj.next().charAt(0);
        }

        System.out.print("Enter character to find its occurance : ");
        char ch = sobj.next().charAt(0);
 
        MyArray mobj = new MyArray();

        int iRet = 0;

        iRet = mobj.Search(Arr, ch);

        System.out.println("Total Capital Characters : " + iRet);
    }
}
